src-input/duk_bi_symbol.c

```diff
@@ -50,7 +50,7 @@ DUK_INTERNAL duk_ret_t duk_bi_symbol_constructor_shared(duk_hthread *thr) {
 		if (++thr->heap->sym_counter[0] == 0) {
 			thr->heap->sym_counter[1]++;
 		}
-		p += DUK_SPRINTF((char *) p, "\xFF" "%lx-%lx",
+		p += DUK_SNPRINTF((char *) p, 1 + 17 + 1 /* 1 + len lost above */, "\xFF" "%lx-%lx",
```

Comment:
This seems incorrect considering C99 snprintf() NUL behavior: if the entire buffer is filled, byte for byte, the last character will actually become a NUL and be lost (the sprintf() variant doesn't have this issue because there's no forced NUL termination).

The fix here would be to add +1 for the (unnecessary) snprintf() NUL terminator.

Reply:
After analyzing this constructor this is my conclusion:

```
A  len B  f  f  f  f  f  f  f  f  -  f  f  f  f  f  f  f  f  C
-- --- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
00 len 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15 16 17 18 19
```
`snprintf` would write in the range B upto (and possibly including C if it writes `NUL`). However, C would be overwritten with `0xff` if `desc` is `NULL`, therefore we don't need to add another byte to the buffer. `snprintf` would return the ideal number of bytes regardless.  

Thus, it's not necessary to change anything in this constructor. I'll add a `DUK_ASSERT` as a check against negative return values, or suddenly asking for more space than the assumption.